cmd/juju/application/upgradecharm.go

```diff
@@ -753,10 +753,8 @@ func detectDefaultSpace(modelConfig *config.Config, curBindings map[string]strin
 			return defaultSpace
 		}
 	}
-
-	// TODO(achilleasa): When we introduce a model config parameter for
-	// specifying the default space we should check whether it is set and
-	// use its value as the default before falling-back to the global
-	// default below.
+	if val := modelConfig.DefaultSpace(); val != "" {
```

Comment:
This shouldn't be in this patch. It will come later.
Before we do this, we will need to wire up use of the model config value as a real space.

Reply:
Ah sorry, my bad. I thought we wanted this in. @nammn Can you please create a card so we don't forget about this change in the future?

Reply:
sure @achilleasa